src/parser/paladin/retribution/modules/core/GlobalCooldown.js

```diff
@@ -0,0 +1,30 @@
+import CoreGlobalCooldown from 'parser/shared/modules/GlobalCooldown';
+import SPELLS from 'common/SPELLS';
+import Haste from './Haste';
+import Abilities from '../Abilities';
+
+const MIN_GCD= 750;
+
+/**
+ * The Global Cooldown from Wake of Ashes is currently double dipping haste
+ */ 
+
+class GlobalCooldown extends CoreGlobalCooldown {
+  static dependencies = {
+    abilities: Abilities,
+    haste: Haste,
+  };
+
+  getGlobalCooldownDuration(spellId) {
+    const gcd = super.getGlobalCooldownDuration(spellId);
+    if (!gcd) {
+      return 0;
+    }
+    if (spellId && spellId === SPELLS.WAKE_OF_ASHES_TALENT.id) {
+      return Math.max(gcd / (1 + this.haste.current));
```

Comment:
This Math.max does nothing?

Reply:
Please leave a comment to explain this change

Reply:
I'm assuming you mean the Math.max in the final return statement and not the nested Math.max that is shown here?

Reply:
no

Reply:
This one has only 1 parameter, so it does nothing?

Reply:
ah, my bad, it needs MIN_GCD as second parameter